edge-hub/src/Microsoft.Azure.Devices.Routing.Core/endpoints/statemachine/EndpointExecutorFsm.cs

```diff
@@ -484,7 +485,7 @@ static async Task EnterProcessDeadAsync(EndpointExecutorFsm thisPtr)
                     // this router to make progress.
                     ICollection<IMessage> tocheckpoint = thisPtr.currentSendCommand.Messages;
                     Events.Dead(thisPtr, tocheckpoint);
-                    SendFailureDetails persistingFailureDetails = thisPtr.currentCheckpointCommand?.Result?.SendFailureDetails.GetOrElse(null);
+                    SendFailureDetails persistingFailureDetails = thisPtr.currentCheckpointCommand?.Result?.SendFailureDetails.GetOrElse(default(SendFailureDetails));
```

Comment:
There may be a `GetOrDefault` call on `Option`?

Reply:
I will check. What happened here is that the version from Edge.Utils had a generic version of GetOrElse (while the one from Routing.Core.Utils not), and when I started using Edge.Utils, the compiler was not happy with the type-less null anymore. It was just a mechanical rewrite to add type info for the compiler to make it happy.